sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,393 @@
+- Fixed the behavior of `static|dynamic_address_cast` when target address space is `generic`. intel/llvm#15394
+- Fixed behaviour of the queue barrier with waitlist for the in-order lists mode in the L0 adapter. intel/llvm#15404
+- Fixed in-order queue dependencies for no-scheduler path when multiple queues are used. intel/llvm#15412
+- Fixed queue barrier with waitlist when used with interoperability events. intel/llvm#15488
+- Fixed edge cases for `exp(complex)` in device code. intel/llvm#15489  intel/llvm#15672 intel/llvm#15980 intel/llvm#15808
+- Inlined trivial util functions for `half`/`half2` in intel math header to avoid multiple definition linker error. intel/llvm#15518
+- Fixed queue barrier with waitlist  for the case when there are multiple queues targetng different devices and in-order command lists are enabled on L0 adapter. intel/llvm#15516
+- Fixed a thread pool data race during shutdown. intel/llvm#15535
+- Fixed UR error handling in CUDA and HIP adapters to prevent uncaught exceptions from leaking out of adapters. intel/llvm#15568
+- Fixed the L0 adapter to properly handle event dependencies for copy commands which are used for bufffer initialization to ensure that buffers contain correct values. intel/llvm#15559
+- Fixed ESIMD `load_2d` inconsistency when reading non-native types with VNNI transforms. intel/llvm#15584
+- SYCL `assert` headers now explicitly include C++ linkage specifications to prevent compilation failures when wrapped in C linkage specifications by applications. intel/llvm#15570 intel/llvm#15614
+- Fixed L0 adapter to reference-count the parent buffer during sub-buffer creation to prevent premature freeing until all sub-buffers are released. intel/llvm#15480
+- Made internal function to be static in `imf` rounding utils to resolve multiple definition errors when developers use `fp32` and `fp64` intel math function in the same compiling unit. intel/llvm#15548
+- Implemented missing work group collectives in Native CPU libdevice. intel/llvm#15618
+- The `max_num_work_groups` query handling for exceeded launch limits has been improved for the HIP and OpenCL backends. intel/llvm#15369
+- Fixed an issue where barrier operations in SYCL Graph did not correctly record dependencies from graph events on another in-order queue. intel/llvm#15601
+- Fixed subgroup read/write implementation for Native CPU. intel/llvm#15627
+- The ESIMD `mask_expand_load` now uses `passthrough` to handle unread elements, fixing sporadic failures. intel/llvm#15664
+- Fixed the issue where  commands like `memcpy`, `copy`, and `fill` were not marked as enqueued because they bypass the  scheduler, causing subsequent barrier commands to be omitted. intel/llvm#15697
+- Fixed `bfloat16` component type matrix `muladd`. intel/llvm#15514
+- Fixed SYCL Graph API to be ABI-neutral to avoid dual-abi issues on Linux. intel/llvm#15694
```

Comment:
Should be moved into API/ABI breaking changes section as an upcoming breaking change